tests/Unit/EuroTest.php

```diff
@@ -10,62 +10,65 @@
 
 /**
  * @covers \WMDE\Euro\Euro
- *
- * @license GPL-2.0-or-later
- * @author Jeroen De Dauw < jeroendedauw@gmail.com >
  */
 class EuroTest extends TestCase {
 
 	/**
 	 * @dataProvider unsignedIntegerProvider
 	 */
-	public function testGetCentsReturnsConstructorArgument( int $unsignedInteger ) {
+	public function testGetCentsReturnsConstructorArgument( int $unsignedInteger ): void {
 		$amount = Euro::newFromCents( $unsignedInteger );
 		$this->assertSame( $unsignedInteger, $amount->getEuroCents() );
 	}
 
-	public function unsignedIntegerProvider() {
+	/**
+	 * @return array{int}[]
```

Comment:
Interesting, never seen array shapes used like this. Probably `array<array<int>>` is clearer tho

Reply:
Or `int[][]`

Reply:
`array{int}` is an [array shape](https://phpstan.org/writing-php-code/phpdoc-types#array-shapes) that signifies a tuple with one `int` element, whereas `array<int>` is saying "this a list (arbitrary length) of integers" (same as `int[]`). `array{int}[]` means a list of these tuples.

In this case it doesn't matter (esp. for PHPStan), but where data providers return mixed tuples (e.g. `array{int,string}[]`) it is useful. 

I also like the precision of tuples that shows the exact number of elements in the array. Not in the language itself (yet), but at least as a PHPDdoc comment that the IDE and static analysis can interpret.